Sourcery/Generating/Template/Stencil/StencilTemplate.swift

```diff
@@ -85,6 +92,15 @@ extension Annotated {
 }
 
 extension Stencil.Extension {
+    
+    func registerFilterWithTwoArguments<A, B>(_ name: String, filter: @escaping (Any?, A, B) throws -> Any?) {
+        registerFilter(name) { (any, args) throws -> Any? in
+            guard args.count == 3, let argA = args[0] as? A, let argB = args[2] as? B else {
+                throw TemplateSyntaxError("'\(name)' filter takes two arguments: \(A.self) and \(B.self)")
+            }
+            return try filter(any, argA, argB)
+        }
+    }
```

Comment:
You can simplify your code:
```swift
func registerFilterWithTwoArguments<T, A, B>(_ name: String, filter: @escaping (T, A, B) throws -> Any?) {
        registerFilter(name) { (any, args) throws -> Any? in
            guard let type = any as? T else { return any } 
            guard args.count == 3, let argA = args[0] as? A, let argB = args[2] as? B else {
                throw TemplateSyntaxError("'\(name)' filter takes two arguments: \(A.self) and \(B.self)")
            }
            return try filter(type, argA, argB)
        }
   }
```

Then at your call site you can get rid of any guess work
```swift
 ext.registerFilterWithTwoArguments("replace", filter: { (source: String, substring: String, replacement: String) -> Any? in
              return source.replacingOccurrences(of: substring, with: replacement)
         })
```